shelley/chain-and-ledger/executable-spec/src/LedgerState.hs

```diff
@@ -1146,12 +1148,38 @@ createRUpd (BlocksMade b) (EpochState acnt ss ls pp) =
         eta = fromIntegral blocksMade / expectedBlocks
 
 -- | Overlay schedule
+-- This is just a very simple round-robin, evenly spaced schedule.
+-- The real implementation should probably use randomization.
 overlaySchedule
-  :: Set (VKeyGenesis dsignAlgo)
+  :: Epoch
+  -> Set (VKeyGenesis dsignAlgo)
   -> Seed
   -> PParams
   -> Map.Map Slot (Maybe (VKeyGenesis dsignAlgo))
-overlaySchedule = undefined
+overlaySchedule e gkeys _ pp = Map.union active inactive
+  where
+    numActive = floor $ dval * fromIntegral slotsPerEpoch
+    dval = unitIntToRational $ pp ^. d
+    dInv = 1 / dval
+    asc = unitIntToRational $ pp ^. activeSlotCoeff
+
+    toRelativeSlot x = (Duration . floor) (fromRational (x % 1)*dInv)
+    toSlot x = firstSlot e +* toRelativeSlot x
+
+    genesisSlots = [ toSlot x | x <-[0..numActive] ]
+
+    numInactivePerActive = floor (fromRational (asc * (numActive % 1)) :: Float) - 1
```

Comment:
why `Float` ?

Reply:
no good reason, `Rational` would be better, I'll use it instead, thanks!